tests/lib/prepare.sh

```diff
@@ -881,6 +881,33 @@ EOF
         exit 1
     fi
 
+    
+    
+    
+    # download the core20 snap manually from the specified channel for UC20
+    if os.query is-core20; then
+        snap download core20 --channel="$BASE_CHANNEL" --basename=core20
+
+        
+        # we want to download the specific channel referenced by $BASE_CHANNEL, but
+        # if we just seed that revision and $BASE_CHANNEL != $IMAGE_CHANNEL, then 
+        # immediately on booting, snapd will refresh from the revision that is 
+        # seeded via $BASE_CHANNEL to the revision that is in $IMAGE_CHANNEL, so to
+        # prevent that from happening (since that automatic refresh will confuse
+        # spread and make tests fail in awkward, confusing ways), we unpack the snap
+        # and re-pack it so that it is not asserted and thus won't be automatically
+        # refreshed
+        if [ "$IMAGE_CHANNEL" != "$BASE_CHANNEL" ]; then
+            sudo unsquashfs -d core20-snap core20.snap
+            sudo snap pack --filename=core20-repacked.snap core20-snap
```

Comment:
Actually one question - do we have any tests that test core20 refreshes that might stop working with this change?

Reply:
It's a fair question, I looked but couldn't find any actually, which is a bit disappointing, I could have sworn we have spread tests around base snap upgrades but perhaps not for UC20 specifically.